src/engine/Service.cpp

```diff
@@ -504,3 +495,117 @@ std::optional<std::string> Service::idToValueForValuesClause(
       }
   }
 }
+
+// ____________________________________________________________________________
+void Service::precomputeSiblingResult(std::shared_ptr<Operation> left,
+                                      std::shared_ptr<Operation> right,
+                                      bool rightOnly, bool requestLaziness) {
+  AD_CORRECTNESS_CHECK(left && right);
+
+  auto skipSortOperation = [](std::shared_ptr<Operation>& op) {
+    auto children = op->getChildren();
+    if (children.size() == 1) {
+      op = children[0]->getRootOperation();
+    }
+  };
+  skipSortOperation(left);
+  skipSortOperation(right);
+
+  auto a = std::dynamic_pointer_cast<Service>(left);
+  auto b = std::dynamic_pointer_cast<Service>(right);
+
+  if ((rightOnly && !static_cast<bool>(b)) ||
+      (!rightOnly && static_cast<bool>(a) == static_cast<bool>(b))) {
+    return;
+  }
```

Comment:
If both are service requests, you could still in principle compute one of them here, and hope that it's the smaller one. 
At least some comment should be left here. IF in the future we figure out a way to better estimate the result size of a service, we could then use the smaller estimate here. But that's not important for now.
As said: A comment why we currently don't support two services should be there at least.